girder/models/user.py

```diff
@@ -244,26 +249,43 @@ def createUser(self, login, password, firstName, lastName, email,
             'groupInvites': []
         }
 
-        self.setPassword(user, password)
+        self.setPassword(user, password, save=False)
+
+        self.setPublic(user, public, save=False)
+
+        # An id is required in order to grant the user access on itself, but
+        # since the user isn't fully created yet, it should not be saved
+        user['_id'] = ObjectId()
+        self.setUserAccess(user, user, level=AccessType.ADMIN, save=False)
+
+        self.save(user)
```

Comment:
We should reassign the return value of `save` back into `user`; save is allowed to modify the values of the document, but the contract does not guarantee that it will do so in-place. It *does* guarantee that it will return the modified document, though.